protocol/flows/relaytransactions/shared_requested_transactions.go

```diff
@@ -37,7 +37,7 @@ func (s *SharedRequestedTransactions) addIfNotExists(txID *daghash.TxID) (exists
 	return false
 }
 
-// NewSharedRequestedTransactions returns a new instance of SharedRequestedTransactions.
+// NewSharedRequestedTransactions returns a new instance of *SharedRequestedTransactions.
```

Comment:
Whether you return it by reference or by value is probably redundant to specify in a comment.